xarray/core/dataarray.py

```diff
@@ -1185,6 +1185,37 @@ def fillna(self, value):
         out = ops.fillna(self, value)
         return out
 
+    def interpolate_na(self, dim=None, method='linear', inplace=False,
+                       **kwargs):
+        """Interpolate values according to different methods.
+
+        Parameters
+        ----------
+        dim : str
+            Specifies the dimension along which to interpolate.
+        method : {'linear', 'time', 'index', 'values', 'nearest'}
+            'linear': ignore the index and treat the values as equally
+                      spaced. default
+            'time': interpolation works on daily and higher resolution data to
+                    interpolate given length of interval
+            'index', 'values': use the actual numerical values of the index
+            'nearest', 'zero', 'slinear', 'quadratic', 'cubic', 'barycentric',
+            'polynomial' is passed to scipy.interpolate.interp1d with the
+            order given both 'polynomial' and 'spline' require that you also
+            specify and order (int) e.g. da.interpolate_na(method='polynomial', order=4)
+
+        # TODO: Pandas also support: scipy wrappers of 'krogh',
+        #       'piecewise_polynomial', 'spline', and 'pchip'
```

Comment:
Personally, I find this interface from pandas a little confusing. It combines two separate choices in one argument:
- Choosing the points at which interpolation is done.
- How to interpolate between these points.

I would slightly rather that we had two separate arguments. Maybe:
- `points='coordinate'` vs `points='equal_spacing'` or a boolean `use_coordinate=True`/`use_coordinate=False`
- `method='nearest'`, `method='linear'`, `method='quadratic'`, etc, for indicating how to interpolate.

The downside is that this would be inconsistent with pandas. But in the existing interface, it is not at all obvious to me whether a method like `'polynomial'` uses coordinate points or equally spaced points.

Reply:
I totally agree. As I've been working on this I've been thinking we can improve on the Pandas interface and this would be the obvious way to do that.